jsonnet/jsonnet.bzl

```diff
@@ -68,7 +77,7 @@ def _jsonnet_library_impl(ctx):
   """Implementation of the jsonnet_library rule."""
   depinfo = _setup_deps(ctx.attr.deps)
   sources = depinfo.transitive_sources + ctx.files.srcs
-  imports = depinfo.imports + ctx.attr.imports
+  imports = depinfo.imports + ctx.attr.imports + [src.dirname for src in ctx.files.srcs]
```

Comment:
If this adds a -J for every source file mentioned in the jsonnet_library srcs attribute, then that is probably dangerous.  You'll end up with file collisions and weird behavior.

Reply:
I wonder if you could just have some import_root=True that would at least add `bazel-out/darwin-fastbuild/path/to/rule/dir`.  So then the paths you import in Jsonnet look like the ones in the srcs attribute of the library rule.

Reply:
That sounds reasonable. Is there a way to query what that root would be?

Reply:
I don't actually know anything about Bazel I'm afraid :)

Reply:
@davidzchen knows things!

Reply:
For context, see last paragraph of PR description